programs/marginfi/src/state/interest_rate.rs

```diff
@@ -115,32 +181,37 @@ impl InterestRateCalc {
             protocol_fee_fixed,
         } = self.get_fees();
 
-        let fee_ir = insurance_fee_rate + group_fee_rate + protocol_fee_rate;
-        let fee_fixed = insurance_fee_fixed + group_fee_fixed + protocol_fee_fixed;
+        let fee_ir: I80F48 = insurance_fee_rate + group_fee_rate + protocol_fee_rate;
+        let fee_fixed: I80F48 = insurance_fee_fixed + group_fee_fixed + protocol_fee_fixed;
 
-        let base_rate_apr = self.interest_rate_curve(utilization_ratio)?;
+        let base_rate_apr: I80F48 = if self.curve_type == INTEREST_CURVE_SEVEN_POINT {
+            self.interest_rate_multipoint_curve(utilization_ratio)?
+        } else {
```

Comment:
Instead of if else, could match LEGACY, SEVEN and error if neither of those. With this case if somehow someone accidentally put in >1 curve type it would default to 0 silently.